packages/protocol/scripts/check-backward.ts

```diff
@@ -104,44 +69,39 @@ try {
     new DefaultCategorizer(),
     out
   )
-  out(`Writing report to ${outFile} ...`)
+  out(`Writing compatibility report to ${outFile} ...`)
   writeJsonSync(outFile, backward, { spaces: 2 })
   out('Done\n')
   if (argv._.includes(COMMAND_REPORT)) {
     // Report always generated
     // Placebo command
-  } else if (argv._.includes(COMMAND_SEM_INFER)) {
-    out(`Inferred version: `)
-    out(
-      backward.report.global.versionDelta
-        .appliedTo(ContractVersion.fromString(argv.old_version))
-        .toString(),
-      true
-    )
-    out(`\n`)
   } else if (argv._.includes(COMMAND_SEM_CHECK)) {
-    const expected = backward.report.global.versionDelta.appliedTo(
-      ContractVersion.fromString(argv.old_version)
-    )
-    if (expected.toString() !== argv.new_version) {
-      out(
-        `${argv.old_version} + ${backward.report.global.versionDelta} != ${argv.new_version}`,
-        true
-      )
-      out(`\n`)
-      process.exit(1)
-    } else {
-      out(
-        `${argv.old_version} + ${backward.report.global.versionDelta} == ${argv.new_version}`,
-        true
-      )
-      out(`\n`)
-      process.exit(0)
+    const doVersionCheck = async () => {
+      const excludeRegexp: RegExp = exclude ? new RegExp(exclude) : null
+      const versionReport = (
+        await ASTContractVersionsReport.create(
+          instantiateArtifacts(oldArtifactsFolder),
```

Comment:
I'm not sure what the concurrency model is in typescript, but since you are already using promises, maybe we could concurrently run both instantiateArtifacts calls? I bet they are probably 80%[citation needed] of the run time of the script

Reply:
It's reasonably performant already, perhaps we save for a future optimization?

Reply:
Sure, not really needed right now